### PR TITLE
Added commons.HandleError() + Fixes and Refactoring

### PR DESCRIPTION
**Pull request by Digitalone1**
* Removed unneeded `break` after `return` in switch (this was reported by Codacy).

Since I saw lots of duplication, I though to add a new endpoint to handle errors, but there are some things to point out.
* I added two `return` which I think they were missing. First is in `DELETEPackagesName` at `!rm.ok` check, the second in `DELETEPackageVersion` at `!pack.ok`. If I was wrong, please remove them and specify with a comment why we don't need to return.
* In case of `No Repo Access`, `common.AuthFail` is called. It expects an object as third argument, but am I wrong or we pass a `content` string? Which I suppose it will default to the case of server error anyway. In the meantime I just added it as non-breaking case over default in `HandleError`. Please check it out if this has to be fixed.

Anyway I think there's also some other duplication in other parts of the code we may reduce implementing helper/utils. I will check it out in the next days when I have time. @confused-Techie If you have any hint on this, let me know.

### CODE REVIEW

File: src/handlers/common_handler.js [hunk 1]
```diff
@@ -124,6 +124,39 @@ async function BadPackageJSON(req, res) {
...
+
+    case "Bad Repo":
+      await BadRepoJSON(req, res);
+      break;
+
+    case "Bad Package":
+      BadPackageJSON(req, res);
+      break;
+
+    case "No Repo Access":
```
Comment (confused-Techie) [inline src/handlers/common_handler.js:152]: Maybe could we have `No Repo Access` resolve to `AuthFail()`, thats what I did within the `package_handler` to give more of a clue why the process failed.

**Review by confused-Techie — CHANGES_REQUESTED** [review]
Love the work here, I've left a comment on one change we should add in. But otherwise this looks great and loved the addition of `HandleError` to abstract away from error handling in the main logic. As for your comments about duplication within the utils, you are absolutely right within `package_handler` theres `DetermineUserPackagePermission` and `DetermineUserPackageGitPermission`, then utils has `LocalUserLoggedIn` which all do very nearly the same thing. The reason is, because I like the benifits of this functions to have less duplication surrounded around handling auth. Although I feel the syntax can be a tad confusing and ugly once implemented. But after implementing `utils` it seems a much better location for this type of function to live, which will be the end goal of where to have it, if used at all. But currently both are used in different places. And its a decision I'm still looking at. 

Additionally on your comment about about gitowner `No Auth` being handled by `AuthFail` I added in an additional argument for `AuthFail` that checks the proper shorts for both user authentication failure, and git authentication failure, so that'll handle it with the `error.MissingAuthJSON()` either way.

Otherwise this looks great, and I only have that one change I would recommend.

**Comment by Digitalone1** [comment]
Check it out. It should be better now.

**Comment by confused-Techie** [comment]
Looks beautiful, as always thanks for the pull request, also thanks for handling the empty JSDoc `@todos` for me. Kept seeing the action failing after I had closed everything down. 

I'll go ahead and merge to main

**Pull request merged**